packages/flutter/lib/src/material/dialog.dart

```diff
@@ -320,11 +355,45 @@ class AlertDialog extends StatelessWidget {
       }
     }
 
-    Widget dialogChild = IntrinsicWidth(
-      child: Column(
-        mainAxisSize: MainAxisSize.min,
-        crossAxisAlignment: CrossAxisAlignment.stretch,
-        children: <Widget>[
+    List<Widget> columnChildren;
+    if (_scrollable) {
+      columnChildren = <Widget>[
+        if (title != null || content != null)
+          Flexible(
+            child: SingleChildScrollView(
+              child: Column(
+                mainAxisSize: MainAxisSize.min,
+                crossAxisAlignment: CrossAxisAlignment.stretch,
+                children: <Widget>[
+                  if (title != null)
```

Comment:
Rather than duplicating the construction of the title and body widgets for both scrolling and non-scrolling cases, it would be better to pull these out into a single place and just include them in the appropriate containers. That way if someone needs to update the title for example, there is only one place to do so.